### PR TITLE
RavenDB-15625 stay in rehab with disabled index

### PR DESCRIPTION
**Pull request by efratshenhar**

### CODE REVIEW

File: src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs [hunk 1]
```diff
@@ -1363,6 +1363,12 @@ private string FindMostUpToDateNode(List<string> nodes, string database, Diction
                     return false;
                 }
 
+                if (mentorIndexStats.State != IndexState.Disabled && currentIndexStats.State == IndexState.Disabled)
+                {
```
Comment (ayende) [inline src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs]: Under what scenario can this happen?

Comment (ayende) [reply src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs]: Also, shouldn't we also check for indexing errors?

Comment (efratshenhar) [reply src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs]: We can change index to disabled only in one node. when that node tries to come back from rehab we don't want to promote him until the index is enabled.
index errors are being handled in https://issues.hibernatingrhinos.com/issue/RavenDB-15588

Comment (ayende) [reply src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs]: That would be a change in the same area of code, right?

Comment (efratshenhar) [reply src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs]: yes

Comment (ppekrol) [reply src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs]: It means that we will stay forever in Rehab in this situation, is this what we want?

Comment (ayende) [reply src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs]: What is the behavior that we _want_ here?

**Review by ppekrol — APPROVED** [review]
Can we rebase this?

**Comment by efratshenhar** [comment]
Done

**Pull request merged**